lib/constants/currency.js

```diff
@@ -1,141 +1,144 @@
-export const Currency = [
-  'USD',
-  'AED',
-  'AFN',
-  'ALL',
-  'AMD',
-  'ANG',
-  'AOA',
-  'ARS',
-  'AUD',
-  'AWG',
-  'AZN',
-  'BAM',
-  'BBD',
-  'BDT',
-  'BGN',
-  'BIF',
-  'BMD',
-  'BND',
-  'BOB',
-  'BRL',
-  'BSD',
-  'BWP',
-  'BYN',
-  'BZD',
-  'CAD',
-  'CDF',
-  'CHF',
-  'CLP',
-  'CNY',
-  'COP',
-  'CRC',
-  'CVE',
-  'CZK',
-  'DJF',
-  'DKK',
-  'DOP',
-  'DZD',
-  'EGP',
-  'ETB',
-  'EUR',
-  'FJD',
-  'FKP',
-  'GBP',
-  'GEL',
-  'GIP',
-  'GMD',
-  'GNF',
-  'GTQ',
-  'GYD',
-  'HKD',
-  'HNL',
-  'HRK',
-  'HTG',
-  'HUF',
-  'IDR',
-  'ILS',
-  'INR',
-  'ISK',
-  'JMD',
-  'JPY',
-  'KES',
-  'KGS',
-  'KHR',
-  'KMF',
-  'KRW',
-  'KYD',
-  'KZT',
-  'LAK',
-  'LBP',
-  'LKR',
-  'LRD',
-  'LSL',
-  'MAD',
-  'MDL',
-  'MGA',
-  'MKD',
-  'MMK',
-  'MNT',
-  'MOP',
-  'MRO',
-  'MUR',
-  'MVR',
-  'MWK',
-  'MXN',
-  'MYR',
-  'MZN',
-  'NAD',
-  'NGN',
-  'NIO',
-  'NOK',
-  'NPR',
-  'NZD',
-  'PAB',
-  'PEN',
-  'PGK',
-  'PHP',
-  'PKR',
-  'PLN',
-  'PYG',
-  'QAR',
-  'RON',
-  'RSD',
-  'RUB',
-  'RWF',
-  'SAR',
-  'SBD',
-  'SCR',
-  'SEK',
-  'SGD',
-  'SHP',
-  'SLL',
-  'SOS',
-  'SRD',
-  'STD',
-  'SZL',
-  'THB',
-  'TJS',
-  'TOP',
-  'TRY',
-  'TTD',
-  'TWD',
-  'TZS',
-  'UAH',
-  'UGX',
-  'UYU',
-  'UZS',
-  'VND',
-  'VUV',
-  'WST',
-  'XAF',
-  'XCD',
-  'XOF',
-  'XPF',
-  'YER',
-  'ZAR',
-  'ZMW',
-];
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+export const Currency = {
+  USD: { fullName: <FormattedMessage defaultMessage="United States Dollar" /> },
+  AED: { fullName: <FormattedMessage defaultMessage="United Arab Emirates Dirham" /> },
+  AFN: { fullName: <FormattedMessage defaultMessage="Afghanistan Afghani" /> },
+  ALL: { fullName: <FormattedMessage defaultMessage="Albania Lek" /> },
+  AMD: { fullName: <FormattedMessage defaultMessage="Armenia Dram" /> },
```

Comment:
`FormattedMessage` returns a react component, not a string. This reduces the possibility of what we can do with this.

We usually use `defineMessages` like [here](https://github.com/opencollective/opencollective-frontend/blob/enhancement/expense-form-invite-validations/lib/i18n/expense.js#L7) for these cases. Then you can add a small helper like `i18nCurrencyName` and just use it whenever you need to display the currency.

But anyway, I don't think we need to re-invent the wheel and ask our translators to work on this, we can rely on standards here:

```es6
const i18nCurrencyName = (intl, currency) => {
  const currencyNames = new Intl.DisplayNames(intl.locale, { type: 'currency' });
  return currencyNames.of(currency);
}
```

Reply:
Thanks for this idea, this greatly simplifies the code. 🤔